packages/gatsby-theme-store/src/components/Error/ErrorHandler.tsx

```diff
@@ -18,14 +18,25 @@ export const handleError = ({ error, errorId }: Props) => {
   console.error(error)
   console.error(errorId)
 
+  const isUserOffline = !window.navigator.onLine
+
   // prevent infinite loop
   if (
     window.location.pathname.startsWith('/404') ||
-    window.location.pathname.startsWith('/500')
+    window.location.pathname.startsWith('/500') ||
+    window.location.pathname.startsWith('/offline')
   ) {
     return
   }
 
+  if (isUserOffline) {
+    const previousPagePath = window.location.pathname
+
+    window.location.href = `/offline?from=${previousPagePath}`
```

Comment:
Do we need to encodeURIComponent this variable?

Reply:
Nice catch! I guess that's a good idea to avoid problems in the future. We might have some weird characters in different locales.